### ISSUE TITLE
`--sort-reexports` does not respect `--profile=black`

### ISSUE THREAD

**Issue by Helveg** [PyCQA/isort#2280]
### Problem

Attempting to sort a multiline `__all__` attribute that is longer than the line limit with `isort . --sort-reexports --profile=black` wraps the lines in a non-black compatible way:

#### Before

```
__all__ = [
    "ArgumentReference",
    "ConfigurationSource",
    "HeptoFlow",
    "HeptoFlowContainer",
    "InMemoryCache",
    "Input",
    "InputReference",
    "InputValue",
    "Service",
    "parse_input_expression",
]
```

#### After

```
__all__ = ['ArgumentReference', 'ConfigurationSource', 'HeptoFlow', 'HeptoFlowContainer',
 'InMemoryCache', 'Input', 'InputReference', 'InputValue', 'Service',
 'parse_input_expression']
```

### Env

```
Python 3.12
isort 5.13.2
black 24.4.2
```

PS: Very willing to fix this myself and to PR it if someone can point me to the right approach.

**Comment by Helveg** [comment]
I did some digging and it seems that code sorting (including the quasi-undocumented and quasi-untested sort comments `# isort: list`, `# isort: unique-list`, etc) all use `isort.literal.assigment`, which relies on `pprint`, and completely ignores all of the `config`, except for `line_length`:

```
class ISortPrettyPrinter(PrettyPrinter):
    """an isort customized pretty printer for sorted literals"""

    def __init__(self, config: Config):
        super().__init__(width=config.line_length, compact=True)
```

There is no regard for the `config.multi_line_output`, `config.include_trailing_comma`, `config.split_on_trailing_comma`, and likely many more that I am not aware of as a first-time contributor.

I believe fixing this would require us to write our own formatting logic, since `pprint` does not support any of the configurable formatting features `isort` needs.

I would need the input of the developers here, since there are no tests to form a de facto specification, but I believe that code sorting should follow the `Config` formatting options, and the `pprint` logic should be rewritten.

**Comment by Helveg** [comment]
@timothycrosley Hi, sorry for the direct ping, I'm sure you're very busy! Would you have a moment to confirm my suspicions that the `code_sorting` logic is insufficient, and preapprove my attempts to rewrite it?

I already have a PR open for fixing similar `sort_reexport` problems: https://github.com/PyCQA/isort/pull/2283

I would require a little bit of dev attention so that that PR can be merged, and I can start work on figuring out the rest of the code sorting issues.

Hope to hear from you, and have a good day

**Comment by lord-haffi** [comment]
Hi @Helveg, just kindly asking, is there any update on this issue?

**Comment by Helveg** [comment]
Yes, this was merged and released in version 6. Can you let me know if it works as expected in v6 for you? If so, I'll close the issue

**Comment by lord-haffi** [comment]
Unfortunately not. I used isort 6.0.1 but this exact issue still persists. Does it work for you? Maybe I'm doing something wrong..

**Comment by staticdev** [comment]
@Helveg Timothy is currently not maintaining isort, but feel free to propose a fix in case you feel like, me and other contributors are applying latest changes.

**Comment by Helveg** [comment]
Oh ok, sorry @lord-haffi I must've gotten my wires crossed :) If I have a quiet weekend I can take a look at this

**Comment by lord-haffi** [comment]
No worries bud, I was just curious :P Take your time :)
I'd do it myself but when I took a look at the codebase... Well, let's say, it'd cost me more than a quiet weekend to work on it ^^

**Comment by lord-haffi** [comment]
Hi, is there any update on this? Just tried it again with isort version 7 and the issue remains. If you can guide me, where I have to look, I could also try it myself.
